primary.c

```diff
@@ -2,29 +2,45 @@
 #include<math.h>
 #include<stdbool.h>
 #include<string.h>
-#include <stdlib.h>
-
-bool isPrime(int number){
-	for (int i = 2; i <= sqrt(number); i++)
-	{
-		if (number % i == 0)
-			return false;
-	}
-	return true;
-}
+#include<stdlib.h>
+#include<sys/types.h>
+#include<unistd.h>
 
 int main(int argc, char *argv[]){
-	if(argc == 1){
-		printf("You don't type any number, Please type numberes throw argv");
-		return 0;
-	}
-	int num;
-	int i = 0;
-	for(i = 1; i < argc; i++){
-		num = atoi(argv[i]);
-		if(isPrime(num)){
-			printf("%d ", num);
-		}
-	}
-	return 0;
+        int pipe_fd[2];
+        pid_t p;
+
+        if (pipe(pipe_fd) == -1){
+                fprintf(stderr, "Pipe Failed");
+                return 1;
+        }
+
+        p = fork();
+        if (p < 0){
+                        fprintf(stderr, "Fork Failed");
+                        return 1;
+        }
+        else if(p == 0){
+                close(pipe_fd[0]);
+                if (argc == 1){
+                        printf("You don't type any number, Please type numberes through argv");
+                        close(pipe_fd[1]);
+                        return 0;
+                }
+                dup2(1, pipe_fd[1]);
+                close(pipe_fd[1]);
+		execve("./is_real_prime", argv, NULL);
+        }
+        else{
+                close(pipe_fd[1]);
+                int i = 0;
+                char number[100];
+                for(i = 1; i < argc; i++){
+                        read(pipe_fd[0], number, 100);
```

Comment:
I want you to check the return of `read`.
Search for `"c EOF read"` to understand why